### PR TITLE
Search bar

### PR DESCRIPTION
**Pull request by trentontri**
**Do not merge**
This PR is currently a work in progress and has been created for previewing the search bar UI.

### CODE REVIEW

File: src/ResultsTable.js [hunk 1]
```diff
@@ -9,13 +9,17 @@ class ResultsTable extends Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    this.setState({ data: nextProps.data });
+  }
```
Comment (chadwithuhc) [inline src/ResultsTable.js]: In general we say not to mimic the data from props to state unless it's changed. If it stays the same, why can't we refer to the props version instead?

File: src/SearchBar.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  };
+
+  // Teach Autosuggest how to calculate suggestions for any given input value.
+  getSuggestions = value => {
+    const inputValue = value.trim().toLowerCase();
+    const inputLength = inputValue.length;
+
+    return inputLength === 0 ? [] : this.props.runners.filter(runner =>
+      runner["First Name"].toLowerCase().slice(0, inputLength) === inputValue
+    );
```
Comment (chadwithuhc) [inline src/SearchBar.js]: Will you add searching of Last Name as well?

File: src/SearchFilters.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  render() {
+    return(
+      <div className="filters">
+        <p>filter by:</p>
+        <a>all runners</a>
+        <a>St. Jude Heroes</a>
+        <a>Boston Qualifiers</a>
+        <a>Age group winners</a>
```
Comment (chadwithuhc) [inline src/SearchFilters.js]: These should be `<button>` unless you want to change the URL during filtering. I think it'd be nice to have a hardcoded URL to share with friends.

File: src/ResultsTable.js (hunk 1 above)
Comment (trentontri) [reply src/ResultsTable.js]: @chadwithuhc good to know! I refactored this here: https://github.com/trentontri/marathon/commit/e2c0d4eeb207d2c09dda01879b09ced6e7968b08

**Pull request merged**